### ISSUE TITLE
PHP requirement 5.4

### ISSUE THREAD

**Issue by francisalfanta** [chriskacerguis/codeigniter-restserver#514]
I am currently using PHP 5.3 in a  QNAP TS-453 Pro with built-in webserver. Upgrading this device PHP version to 5.4 is not supported as far as I've read the QNAP forum.

Since you mention PHP requirement 5.4, is it possible I can still used PHP 5.3? OR
What part of codeigniter-resterver will break if I used the PHP 5.3?

**Comment by softwarespot** [comment]
One thing that comes to mind is you will need to change array short syntax from [] to array().

As I have only started using PHP from v5.6, I advise you to check the source code and differences between PHP 5.3 and 5.4.

**Comment by ivantcholakov** [comment]
IMO right now switching to PHP 5.4 syntax does not give value. I support in my place PHP 5.3 compatible code. Probably it would run on PHP 5.2.x, but I am lazy to test that.

If you decide to downgrade the code for PHP 5.3, I can do this.

**Comment by softwarespot** [comment]
It would make sense to be on par with CI3 too. I just tried to check with Netbeans using 5.2 as the minimum, but it let [] pass, which it shouldn't. So I won't be tackling this one I'm afraid.

**Comment by ivantcholakov** [comment]
I have a proposal how it could be done.

When all the current pull requests are cleared, we can easily downgrade code for PHP 5.2.4. Then a new release 2.7.2 is to be done. This would be the last release that supports PHP 5.2.4.

Immediately after that code is to be switched to PHP 5.4 syntax again and a new release 2.8.0 with the same features is to be done with minimum required version PHP 5.4. And then, next releases might be developed without bothering about the old PHP versions anymore.

**Comment by softwarespot** [comment]
Fair deal I think. I wonder what @chriskacerguis  thinks?

**Comment by lagaisse** [comment]
Please consider that:
* PHP 5.2 reached its end of life more than 4 years ago 
* PHP 5.3 reached its end of life 11 months ago

**Comment by ivantcholakov** [comment]
@lagaisse We all know that.

**Comment by softwarespot** [comment]
As @ivantcholakov said. Personally I wish everyone would move to v5.6, but as this is for CI3 and they support v5.2 as a minimum, we must respect that and ensure that those who still use an old (and buggy) version of PHP, can run this library without any inconsistency.

The best thing that could happen is if the devs of CI3 were to include this library in the next version of CI. How we can do that is by reaching a wider audience and providing a bug-free library. Wishful thinking =)

**Comment by lagaisse** [comment]
I was pointing that people won't find any serious hosting providing 5.2.

@softwarespot fair deal

**Comment by softwarespot** [comment]
Once my next PR goes through (hopefully), I will leave it for one of you to make the changes, as I don't have an old version of PHP and can't really be trusted for this time for making such delicate changes.

**Comment by ivantcholakov** [comment]
I still have a PHP 5.2.17 server, so I can make the tests and to prepare both of the proposed releases. Tell me when the time comes.

**Comment by softwarespot** [comment]
Whenever @chriskacerguis merges my current PR.

**Comment by chriskacerguis** [comment]
@softwarespot I'ver merged it :)

Regarding the version, as per CI 3's docs:

```
PHP version 5.4 or newer is recommended.

It should work on 5.2.4 as well, but we strongly advise you NOT to run such old versions of PHP, because of potential security and performance issues, as well as missing features.
```
http://www.codeigniter.com/user_guide/general/requirements.html

So, that said, I'm not sure that back porting to 5.2, then re-porting it to 5.4+ syntax is a worth wild effort.

**Comment by ivantcholakov** [comment]
@francisalfanta You can find downgraded versions of REST_Controller.php and Format.php within this unofficial project https://github.com/ivantcholakov/codeigniter-restserver-test

**Comment by mckaygerhard** [comment]
some sites can upgrade, so i help with php 5.3 compatible things: https://github.com/codeigniterpower ci power has merged php 5.3 compatible versions of CI-REST -server
